### ISSUE TITLE
[BUG] wrong type in dynamo table query KeyConditionExpression argument

### ISSUE THREAD

**Issue by dpinol** [vemel/mypy_boto3_builder#41]
**Describe the bug**
Argument KeyConditionExpression of dynamo Table.query method has type str but should allow ConditionBase (I use Between to define a range for a sorting key)

**To Reproduce**
1. Install 'boto3-stubs[...]'
2.  Run mypy on the following code sample

```python
import boto3

key_exp = Key("partition_key").eq("pk") & Key("time").between(888888, 999999)

resource: DynamoDBServiceResource = boto3.resource("dynamodb")
resource.table.query(
            KeyConditionExpression=key_exp, Limit=limit
        )
```
3. Actual mypy output

```
src/infrastructure/dynamodb.py:104: error: Argument "KeyConditionExpression" to "query" of "Table" has incompatible type "And"; expected "Optional[str]"  [arg-type]
```

**Expected mypy-output**

No error expected

**Additional context**
boto3 1.14.43

**Comment by vemel** [comment]
Condition Expressions were hard to annotate. They do not use dynamically generated type annotations. Instead a added a static stub: https://github.com/vemel/mypy_boto3_builder/blob/master/mypy_boto3_builder/boto3_stubs_static/dynamodb/conditions.pyi

I will take a look if I can make `&` and `|` operations to return correct type.

**Comment by vemel** [comment]
I just pushed a patch to master, so expect a release later today.

**Comment by vemel** [comment]
Thank you for the report. The bug was fixed and included to the latest release. Please let me know if I overlooked something.
